packages/core/lib/console.js

```diff
@@ -232,9 +264,38 @@ class Console extends EventEmitter {
     const contextVars = {};
 
     abstractions.forEach(abstraction => {
-      contextVars[abstraction.contract_name] = abstraction;
+      // don't overwrite Node's native objects - only load contracts
+      // into the repl context when no conflict exists
+      if (
+        this.replContextNameConflicts[abstraction.contract_name].conflict ===
+        false
+      ) {
+        contextVars[abstraction.contract_name] = abstraction;
+      }
     });
 
+    if (this.warnUserOfConflicts) {
+      let contractNames = [];
```

Comment:
`recordNameConflicts` could build this list, and this function could just use it without having to mine the `replContextNameConflicts` object.